src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let tmpString = '';
+    const lastKey = this.length - 1;
+
+    for (const key of this) {
+      if (key === null || key === undefined) {
+        tmpString += '';
+      } else {
+        tmpString += key;
+      }
+
+      if (key !== this[lastKey]) {
```

Comment:
what if you will have this array:
[1, 2, 3, 4, 2]
how this condition will works?
you will receive string: "1.23.4.2"
original join doesn't works same

Reply:
fix it
as Tetiana said it will work in the wrong way
use common loop `for` instead of `for of` and check index